controllers/k8ssandra/cleanup.go

```diff
@@ -111,14 +96,20 @@ func (r *K8ssandraClusterReconciler) checkDeletion(ctx context.Context, kc *api.
 		if r.deleteCronJobs(ctx, kc, dcTemplate, namespace, remoteClient, logger) {
 			hasErrors = true
 		}
+
+		if dcHasErrors, earlyReturn := r.deleteCassandraDatacenter(ctx, dcTemplate, namespace, remoteClient, logger); earlyReturn != nil {
+			return *earlyReturn
```

Comment:
issue: Why is this using named parameters and a pointer representation of interface?